arch/arm/boot/dts/overlays/ov7251-overlay.dts

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: GPL-2.0-only
+// Definitions for OV7251 camera module on VC I2C bus
+/dts-v1/;
+/plugin/;
+
+#include <dt-bindings/gpio/gpio.h>
+
+/{
+	compatible = "brcm,bcm2835";
+
+	fragment@0 {
+		target = <&i2c_csi_dsi>;
+		__overlay__ {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			status = "okay";
+
+			ov7251: ov7251@60 {
+				compatible = "ovti,ov7251";
+				reg = <0x60>;
+				status = "okay";
+
+				clocks = <&ov7251_clk>;
+				clock-names = "xclk";
+				clock-frequency = <24000000>;
+
+				vdddo-supply = <&ov7251_dovdd>;
+				vdda-supply = <&ov7251_avdd>;
+				vddd-supply = <&ov7251_dvdd>;
+
+				enable-gpios = <&gpio 41 GPIO_ACTIVE_HIGH>;
+
+				port {
+					ov7251_0: endpoint {
+						remote-endpoint = <&csi1_ep>;
+						clock-lanes = <0>;
+						data-lanes = <1>;
+						clock-noncontinuous;
+						link-frequencies =
+							/bits/ 64 <456000000>;
+					};
+				};
+			};
+		};
+	};
+
+	fragment@1 {
+		target = <&csi1>;
+		__overlay__ {
+			status = "okay";
+
+			port {
+				csi1_ep: endpoint {
+					remote-endpoint = <&ov7251_0>;
+					data-lanes = <1>;
```

Comment:
Some of the camera overlays give the `csi` endpoint a `clock-lanes` property, while others don't. What's the logic behind the choice?

Reply:
```clock-lanes``` ought to be there for completeness, but isn't mandatory in our case.
Some SoCs (and a very few sensors) allow reassigning of CSI pair to function, so you might get
```
clock-lanes = <4>
data-lanes = <0 2 1 3>
```
if someone had been particularly perverse in their wiring.

Unicam doesn't support reordering, so ```clock-lane``` will always be <0> (assumed and never checked), and the only thing that matters in ```data-lanes``` is the number of entries. I did put in ```clock-lane``` for completeness on a couple of overlays, and probably ought to here too.

The other parameter that might be worth specifying is the ```bus-type``` (CSI-2 DPHY vs CCP2 vs parallel, etc). Upstream are moving towards pushing this explicitly into DT instead of deducing it from the other parameters that are specified on the endpoint. I may blitz them all if I get particularly bored.